File: aisile-sellergoods-interface/src/main/java/com/aisile/sellergoods/service/TypeTemplateService.java
package com.aisile.sellergoods.service;

import java.util.List;
import java.util.Map;

import com.aisile.pojo.TbTypeTemplate;
import com.aisile.pojo.entity.PageResult;

public interface TypeTemplateService {
	/**
	 * 查询全部数据
	 * @return
	 */
	public List<TbTypeTemplate> findAll();
	/**
	 * 分页
	 * @param pageNum
	 * @param pageSize
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	/**
	 * 模糊查询
	 * @param pageNum
	 * @param pageSize
	 * @param tbTypeTemplate
	 * @return
	 */
	public PageResult findSearch(int pageNum,int pageSize,TbTypeTemplate tbTypeTemplate);
	/**
	 * 添加
	 * @param tbTypeTemplate
	 */
	public void add(TbTypeTemplate tbTypeTemplate);
	/**
	 * 修改
	 * @param tbTypeTemplate
	 */
	public void update(TbTypeTemplate tbTypeTemplate);
	/**
	 * 回显    返回的是一个对象
	 * @param id
	 * @return
	 */
	public TbTypeTemplate findOne(Long id);
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long [] ids);
	/**
	 * 规格选项使用
	 */
	public List<Map> findOptionsList(Long id);
}

File: aisile-shop-web/src/main/webapp/js/service/typeTemplateService.js
//封装所有连接后端代码    需要$http
app.service('typeTemplateService',function($http){
	//查询全部的数据
	this.findAll = function(){
		return $http.get('../typeTemplate/findAll.do');
	}
	//分页
	this.findPage = function(page,rows){
		return $http.get('../typeTemplate/findPage.do?page='+page+"&rows="+rows);
	}
	//分页带数据查询
	this.findSearch = function(page,rows,searchEntity){
		return $http.post('../typeTemplate/findSearch.do?page='+page+"&rows="+rows,searchEntity);
	}
	//修改
	this.update = function(entity){
		return $http.post('../typeTemplate/update.do',entity);
	}
	//添加
	this.add = function(entity){
		return $http.post('../typeTemplate/add.do',entity);
	}
	//根据id查询对象
	this.findOne = function(id){
		return $http.post('../typeTemplate/findOne.do?id='+id);
	}
	//删除
	this.dele = function(selectIds){
		return $http.get('../typeTemplate/dele.do?ids='+selectIds);
	}
	//查询规格列表
	this.findSpecList = function(id){
		return $http.get('../typeTemplate/findSpecList.do?id='+id);
	}
})